Update view logs page for new features (#2122)
astronomer/docs: astro/view-logs.md

title: 'View logs'
sidebar_label: 'View logs'
id: view-logs
description: View logs for your data pipelines both on Astro and locally.
---

View task and component logs for your DAGs to troubleshoot your data pipelines and better understand the behavior of your tasks and their execution environment.

## Log and message types

> - _Scheduler logs_ can help you understand scheduler performance and indicate if a task failed due to an issue with the scheduler. For more information on configuring the scheduler on Astro, see [Scheduler resources](configure-deployment-resources.md#scheduler-resources).

Reviewer: ```suggestion
- _Scheduler logs_ describe the performance of the scheduler, which is responsible for scheduling and queueing task runs. For more information on configuring the scheduler on Astro, see [Scheduler resources](configure-deployment-resources.md#scheduler-resources).
```

> - _Triggerer logs_ provide information about the performance of the triggerer, the Airflow component responsible for running triggers and signaling tasks to resume when their conditions have been met. The triggerer is used exclusively for tasks that are run with [deferrable operators](/learn/deferrable-operators.md).

Reviewer: ```suggestion
- _Triggerer logs_ describe the performance of the triggerer, the Airflow component responsible for running triggers and signaling tasks to resume when their conditions have been met. The triggerer is used exclusively for tasks that are run with [deferrable operators](/learn/deferrable-operators.md).
```

> - _Triggerer logs_ provide information about the performance of the triggerer, the Airflow component responsible for running triggers and signaling tasks to resume when their conditions have been met. The triggerer is used exclusively for tasks that are run with [deferrable operators](https://docs.astronomer.io/learn/deferrable-operators.md).

Reviewer: ```suggestion
- _Triggerer logs_ describe the performance of the triggerer, the Airflow component responsible for running triggers and signaling tasks to resume when their conditions have been met. The triggerer is used exclusively for tasks that are run with [deferrable operators](https://docs.astronomer.io/learn/deferrable-operators.md).
```

> - _Worker logs_ describe the performance of the _woriker_, the process that executes tasks, as defined by the [executor](/learn/airflow-executors-explained.md#what-is-an-executor)

Reviewer: ```suggestion
- _Worker logs_ describe the performance of your _workers_, including how they execute tasks.
```

> - _Worker logs_ describe the performance of the _woriker_, the process that executes tasks, as defined by the [executor](https://docs.astronomer.io/learn/airflow-executors-explained.md#what-is-an-executor)

Author: @bote795 can maybe elaborate here - but from poking around in Dev, they look the same between the Cloud UI and Airflow UI to me, but would like confirmation on that.

- _Webserver logs_ relate to the health and performance of [the Airflow UI](https://docs.astronomer.io/learn/intro-to-airflow.md#airflow-components)

> ### Log Levels 

Reviewer 2: i think this is fine, though we don't have a log-level column/or filter. As far as my understanding not all logs will show a log level inside the event/log so just want to point that out

---

View task and component logs for your DAGs to troubleshoot your data pipelines and better understand the behavior of your tasks and their execution environment.

## Log and message types

- _Scheduler logs_ describe the performance of the scheduler, which is responsible for scheduling and queueing task runs. For more information on configuring the scheduler on Astro, see [Scheduler resources](configure-deployment-resources.md#scheduler-resources).

- _Triggerer logs_ describe the performance of the triggerer, the Airflow component responsible for running triggers and signaling tasks to resume when their conditions have been met. The triggerer is used exclusively for tasks that are run with [deferrable operators](https://docs.astronomer.io/learn/deferrable-operators.md).

- _Worker logs_ are generated by your workers as they execute tasks. These are the same task logs that you can access directly from the Airflow UI as described in [View Airflow task logs](#view-airflow-task-logs).

- _Webserver logs_ relate to the health and performance of [the Airflow UI](https://docs.astronomer.io/learn/intro-to-airflow.md#airflow-components)

### Log Levels 

> Logs and messages might also be associated with one of the following _log levels_: 

Reviewer 3: this section is uneeded as log levels are no longer being returned

- **Error**: Emitted when a process fails or does not complete. For example, these logs might indicate a missing DAG file, an issue with your scheduler's connection to the Airflow database, or an irregularity with your scheduler's heartbeat.
- **Warn**: Emitted when Airflow detects an issue that may or may not be of concern but does not require immediate action. This often includes deprecation notices marked as `DeprecationWarning`. For example, Airflow might recommend that you upgrade your Deployment if there was a change to the Airflow database or task execution logic.
- **Info**: Emitted frequently by Airflow to show that a standard scheduler process, such as DAG parsing, has started. These logs are frequent but can contain useful information. If you run dynamically generated DAGs, for example, these logs will show how many DAGs were created per DAG file and how long it took the scheduler to parse each of them.

## View logs in the Cloud UI

> You can access scheduler, triggerer, and task logs in the Cloud UI to find the past 24 hours of logs for any Deployment on its **Logs** page. Logs are color-coded according to their type. 

Reviewer 3: "Logs are color-coded according to their type" remove this we do not have log levels anymore so this is not true.

5. Click **Instance Details**.
6. Click **Log**.

## View logs in the Cloud UI

You can access scheduler, triggerer, and worker logs in the Cloud UI to find the past 24 hours of logs for any Deployment on its **Logs** page. Logs are color-coded according to their type. 

1. In the Cloud UI, select a Workspace and then a Deployment.

2. Click the **Logs** tab.

> 3. (Optional) Select one or more of the following options to filter logs and click **Apply**. 

Reviewer 2: no apply button will be available as you select a filter it will update the data.

    When a Deployment generates more than 500 lines of logs in 24 hours, only the most recent 500 lines are shown. If there are no logs available for a given Deployment, the following message appears:

    ```  
    No matching events have been recorded in the past 24 hours.
    ```

    Typically, this indicates that the Deployment you selected does not currently have any DAGs running.

### Filter options

You can use the following options to specify the types of logs or messages that you want to view. 

- **String search**: Enter a string, keyword, or phrase to find in your logs.

> - **Time range**: Filter the logs displayed based on time or a date range. 

Reviewer 2: we wont have date range ready yet so we might want to remove this (it will be on set time intervals, 1hr, 6, 12, 24)

> - **Log type**: Filter based on whether the log message is from scheduler, worker, or trigger. 

Reviewer 2: webserver as well

> - **Log Level**

Reviewer 2: you wont be able to filter by log level after all

5. Click **Instance Details**.
6. Click **Log**.

## View logs in the Cloud UI

You can access scheduler, triggerer, and worker logs in the Cloud UI to find the past 24 hours of logs for any Deployment on its **Logs** page. Logs are color-coded according to their type. 

1. In the Cloud UI, select a Workspace and then a Deployment.

2. Click the **Logs** tab.

>     When a Deployment generates more than 500 lines of logs in 24 hours, only the most recent 500 lines are shown. If there are no logs available for a given Deployment, the following message appears:

Reviewer 2: since we have pagination the max number of items we will get is `10000` but only 25 per page but you can go to the next page to get the next 10.

>     ```  

Reviewer: ```suggestion
    ```text
```

    No matching events have been recorded in the past 24 hours.
    ```

    Typically, this indicates that the Deployment you selected does not currently have any DAGs running.

### Filter options

You can use the following options to specify the types of logs or messages that you want to view. 

- **String search**: Enter a string, keyword, or phrase to find in your logs.

- **Time range**: Filter the logs displayed based on time. 

> - **Log type**: Filter based on whether the log message is from scheduler, worker, webserver, or trigger. 

Reviewer: ```suggestion
- **Log type**: Filter based on whether the log message is from a scheduler, worker, webserver, or trigger. 
```
